### PR TITLE
Fix _serializeKey and _serializeValue

### PR DESCRIPTION
**Pull request by ralphtheninja**
Closes https://github.com/Level/deferred-leveldown/issues/26

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -138,6 +154,74 @@ test('many operations', function (t) {
...
+    }
+    wrapper.idx = -1
+    db[m] = wrapper
+    return new DeferredLevelDOWN(db)
+  }
+
+  function noop () {}
+
+  t.test('put', function (t) {
+    var ld = Db('put', function (key, value, cb) {
```
Comment (vweevers) [inline test.js]: You're essentially spying on the `put` call, correct? Can you simplify it to a pattern like:

```js
var calls = []
doTheThingThatMakesTheCalls()
t.same(calls, [{ key: .., value: ..}, { key: .., value: ..}]
```

Comment (ralphtheninja) [reply test.js]: Correct. Sure!

File: test.js [hunk 2]
```diff
@@ -138,6 +154,88 @@ test('many operations', function (t) {
...
+  })
+
+  function Db (m, fn) {
+    var db = {
+      open: function (options, cb) {
+        process.nextTick(cb)
+      }
+    }
+    var wrapper = function () {
+      wrapper.idx++
```
Comment (vweevers) [inline test.js]: love it, much cleaner. don't need `idx` anymore

**Comment by ralphtheninja** [comment]
Should we look at this as a bug fix? I.e. patch version?

**Comment by vweevers** [comment]
I say yes

**Comment by ralphtheninja** [comment]
2.0.2

**Pull request merged**